Lib/Src/CDispatcher.cpp

```diff
@@ -233,15 +222,17 @@ void CDispatcher::processComChannels()
  * @brief Method to find number of the controller based on the name.
  *
  * @param name Name of the controller.
- * @return
+ * @return Index of the controller if the name was recognised. -1 if not.
  */
-uint8_t CDispatcher::findControllerNumber(etl::string<MAX_STRING_SIZE> name)
+int8_t CDispatcher::findControllerNumber(etl::string<MAX_STRING_SIZE> name)
 {
-    uint8_t controller = 0;
-    while ((m_controller_names[controller].compare(name) != 0) &&
-           (controller < m_controller_count))
+    int8_t controller = -1;
+    for (uint8_t i = 0; i < m_controller_count; i++)
     {
-        controller++;
+        if (m_controller_names[i].compare(name) == 0)
+        {
+            controller = i;
```

Comment:
you can break out of the for loop here if controller is found.
```
controller = i;
break;
```
or you can have `while()` loop instead of `for()` loop:
```
while((controller < 0) && (i < controller_count))
    ...
    ++i;
```

Reply:
Added a break;